sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/SyncRestProxy.java

```diff
@@ -0,0 +1,313 @@
+        if (tracingContext == null) {
+            return;
+        }
+
+        // Get the context that was added to the mono, this will contain the information needed to end the span.
+        Object disableTracingValue = (tracingContext.getData(Tracer.DISABLE_TRACING_KEY).isPresent()
+            ? tracingContext.getData(Tracer.DISABLE_TRACING_KEY).get() : null);
+        boolean disableTracing = Boolean.TRUE.equals(disableTracingValue != null ? disableTracingValue : false);
+
+        if (disableTracing) {
+            return;
+        }
+
+        int statusCode = 0;
+
+        // On next contains the response information.
+        if (httpDecodedResponse != null) {
+            //noinspection ConstantConditions
+            statusCode = httpDecodedResponse.getSourceResponse().getStatusCode();
+        } else if (throwable != null) {
+            // The last status available is on error, this contains the error thrown by the REST response.
+            // Only HttpResponseException contain a status code, this is the base REST response.
+            if (throwable instanceof HttpResponseException) {
+                HttpResponseException exception = (HttpResponseException) throwable;
+                statusCode = exception.getResponse().getStatusCode();
+            }
+        }
+        TracerProxy.end(statusCode, throwable, tracingContext);
+    }
+
+    public void updateRequest(RequestDataConfiguration requestDataConfiguration, SerializerAdapter serializerAdapter) throws IOException {
+        boolean isJson = requestDataConfiguration.isJson();
+        HttpRequest request = requestDataConfiguration.getHttpRequest();
+        Object bodyContentObject = requestDataConfiguration.getBodyContent();
+
+        if (isJson) {
+            ByteArrayOutputStream stream = new AccessibleByteArrayOutputStream();
+            serializerAdapter.serialize(bodyContentObject, SerializerEncoding.JSON, stream);
+
+            request.setHeader("Content-Length", String.valueOf(stream.size()));
+            request.setBody(BinaryData.fromStream(new ByteArrayInputStream(stream.toByteArray(), 0, stream.size())));
+        } else if (bodyContentObject instanceof byte[]) {
+            request.setBody((byte[]) bodyContentObject);
+        } else if (bodyContentObject instanceof String) {
+            final String bodyContentString = (String) bodyContentObject;
+            if (!bodyContentString.isEmpty()) {
+                request.setBody(bodyContentString);
+            }
+        } else if (bodyContentObject instanceof ByteBuffer) {
+            request.setBody(((ByteBuffer) bodyContentObject).array());
+        } else {
+            ByteArrayOutputStream stream = new AccessibleByteArrayOutputStream();
+            serializerAdapter.serialize(bodyContentObject, SerializerEncoding.fromHeaders(request.getHeaders()), stream);
```

Comment:
Instead of using an output stream we could just call `serializerAdapter.serializeToBytes` which may remove duplicating data when writing to the output stream.